### PR TITLE
write logs to stderr instead of stdout fixes #9547

### LINKED ISSUE #9547: logging error should go to stderr instead of stdout
logging module logs errors `error` proc for Console writes to stdout instead of stderror 

### Example
```nim
var L* = newConsoleLogger()
addHandler(L)

error("should got to stderr")
```

### Current Output
stderr is empty
stdout has the message

### Expected Output
stderr has the message
stdout is empty

### Possible Solution
having a special case for errors to stderr instead of stdout
https://github.com/nim-lang/Nim/blob/master/lib/pure/logging.nim#L153

### PR DESCRIPTION
**Pull request by xmonader**
Fixes https://github.com/nim-lang/Nim/issues/9547

write all logs to stderr

```nim
import logging

var L* = newConsoleLogger()
addHandler(L)

error("err should go to stderr")
fatal("fatal sholud go to stderr")
warn("warn sholud go to stderr")
info("should go to stderr")
```

```bash
~> bin/nim -r c /tmp/testlogging.nim > stdoutfile.txt 2> stderr.txt
~> cat stdoutfile.txt  
~> cat stderr.txt
Hint: used config file '/home/ahmed/wspace/Nim/config/nim.cfg' [Conf]
Hint: system [Processing]
Hint: testlogging [Processing]
Hint: logging [Processing]
Hint: strutils [Processing]
Hint: parseutils [Processing]
Hint: math [Processing]
Hint: bitops [Processing]
Hint: algorithm [Processing]
Hint: unicode [Processing]
Hint: times [Processing]
Hint: options [Processing]
Hint: typetraits [Processing]
Hint: strformat [Processing]
Hint: macros [Processing]
Hint: posix [Processing]
Hint: os [Processing]
Hint: ospaths [Processing]
Hint:  [Link]
Hint: operation successful (33727 lines compiled; 0.688 sec total; 48.273MiB peakmem; Debug Build) [SuccessX]
Hint: /tmp/testlogging  [Exec]
ERROR err should go to stderr
FATAL fatal sholud go to stderr
WARN warn sholud go to stderr
INFO should go to stderr
```

### CODE REVIEW

File: lib/pure/logging.nim [hunk 1]
```diff
@@ -150,8 +150,11 @@ method log*(logger: ConsoleLogger, level: Level, args: varargs[string, `$`]) =
       {.emit: "console.log(`cln`);".}
     else:
       try:
-        writeLine(stdout, ln)
-        if level in {lvlError, lvlFatal}: flushFile(stdout)
+        if level in {lvlError, lvlFatal, lvlWarn}:
```
Comment (muhamadazmy) [inline lib/pure/logging.nim]: I think ALL log levels should go to `stderr` by default

Comment (xmonader) [reply lib/pure/logging.nim]: updated (y)

**Comment by Araq** [comment]
stderr is unbuffered and MANY people care about efficient logging and Nim's is already slow enough... Not comparable to Java's or C#'s solutions.

**Comment by cooldome** [comment]
Should we try making it configurable? and keep stdout as a default

**Comment by Araq** [comment]
Yeah, it should be configurable.

**Comment by xmonader** [comment]
@Araq I made it configurable with useStderr 
```nim
import logging

var L* = newConsoleLogger(useStderr=true)
addHandler(L)

error("err should go to stderr")
fatal("fatal sholud go to stderr")
warn("warn sholud go to stderr")
info("should go to stderr")
```

**Pull request merged**